config.js

```diff
@@ -247,6 +250,26 @@ function setRankedUnrankedOptionsDefaults(rankedUnrankedOptions, argv) {
     }
 }
 
+function exportValidLogfileFilename(argvLogfile) {
+    if (typeof argvLogfile === "string") {
+        const validFilename = getValidFilename(argvLogfile);
+        if (argvLogfile !== validFilename) {
+            console.log (`Invalid logfile name "${argvLogfile}" has been automatically renamed`
+                         + ` to "${validFilename}".\nValid logfile name can only be composed of`
+                         + ` upper case (ex: A), lower case (ex: a), numbers, hyphens (-).\n`);
+        }
+        exports.logfile = validFilename;
+
+    } else {
+        // provide default filename based on start_time date and time
+        //
+        const filenameDate = `gtp2ogs-logfile-${exports.start_date.toISOString()}`;
+        const validFilenameDate = getValidFilename(filenameDate);
+        exports.logfile = validFilenameDate;
+        console.log(`Non-string logfile filename was successfully exported as ${validFilenameDate}`)
```

Comment:
Do we really need this else block and this string check?

Also i dont see how this would work if the argument is left out for filename. It will be an empty string then right? But still a string?

I think you mean to run the else block if no filename is supplied? if so, the code can still be refactored, first decided the filename and then put the duplicated instance of `exports.logfile = validFilename;` after the `if`.

Reply:
oh indeed, logfile type should only be string, so we should add a .string in getArgv.js https://github.com/online-go/gtp2ogs/blob/devel/getArgv.js#L26

if only `--logfile` is used without anything, it would be a boolean and not an empty string

but then empty string would be tested false, so the code should slightly be adjusted to suppor that too

yes this can be improved a lot 🎉